src/librustc_typeck/check/compare_method.rs

```diff
@@ -374,7 +374,8 @@ fn compare_predicate_entailment<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
                 &infcx.parameter_environment.caller_bounds);
             infcx.resolve_regions_and_report_errors(&free_regions, impl_m_body_id);
         } else {
-            let fcx = FnCtxt::new(&inh, tcx.types.err, impl_m_body_id);
+            let mut fcx = FnCtxt::new(&inh, impl_m_body_id);
```

Comment:
The majority of cases do not need to recompute it, from what I can tell. Basically 3/4 :). In one of those cases, the default of `None` suffices. So I think it's better to pass it as an argument -- harder to forget that you may want to assign it.